src/pywifes/wifes_wsol.py

```diff
@@ -922,7 +915,7 @@ def xcorr_shift_all( packaged_args ):
                 this_ref_array[numpy.where(this_init_x_array==item)] = \
                     numpy.max(final_lam_bes[loc-2:loc+3])
 
-    return [this_row,this_init_x_array,this_ref_array]
+    return [this_row, this_ref_array]
 

```

Comment:
The new function and the refactoring look fine for me. Also, I've run some tests and they return what expected. I'll take a look at the extracted spectra to double-check that all is ok as some changes affect the wavelength solution and calibration.

Reply:
Perfect. If there are any issues, I can try disabling different parts of the 'multithread' json to track down where the difference occurs.